analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,219 @@
+        telomerase_score = NormEXTENDScores) %>%
+ select(Kids_First_Biospecimen_ID_RNA, telomerase_score)
+
+# remove samples without RNA-Seq since both classifiers' results use RNA-Seq
+# independent use primary plus rna-seq samples
+meta_indep <- tp53_scores %>%
+  left_join(tel_scores) %>%
+  left_join(histologies) %>%
+  dplyr::filter(Kids_First_Biospecimen_ID_RNA %in% indep_rna$Kids_First_Biospecimen_ID) %>%
+  # recode the categories
+  dplyr::mutate(OS_status_fixed = case_when(
+    OS_status == "LIVING" ~ 0,
+    OS_status == "DECEASED" ~1
+ ))
+
+```
+ 
+### Categorical variable cancer group and HGG or not 
+```{r}
+# separate into HGG or non-HGG
+meta_indep <- meta_indep %>% 
+  dplyr::mutate(hgg_group = case_when(
+    cancer_group %in% c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma") ~ "HGAT",
+    !cancer_group %in% c("High-grade glioma astrocytoma", "Diffuse midline glioma", "Diffuse intrinsic pontine glioma") ~ "non-HGAT"
+  ))
+
+# filter to contain only samples in short histology that that have >=10 samples
+meta_indep <- meta_indep %>% 
+  dplyr::group_by(cancer_group) %>% 
+  dplyr::mutate(n_cg = n()) %>% 
+  dplyr::mutate(cancer_group_over10 = case_when(
+    n_cg >= 10 ~ cancer_group
+  ))
+
+```
+
+## Cox Regression 
+### TP53 scores & Telomerase score - results
+```{r Run Cox regression model for tp53 and telomerase for all covariates}
+# run the model with different covariates
+for(ind_var in c("tp53_score", "telomerase_score", "cancer_group_over10", "hgg_group")){
+  model <- paste0("survival::Surv(time = OS_days, event = OS_status_fixed) ~ ", ind_var)
+  
+  fit <- survival::coxph(
+        formula(model),
+        data = meta_indep
+      )
+  # generate output
+  table <- broom::tidy(fit) 
+  
+  # Save the table data in a TSV
+  readr::write_tsv(table, file.path(results_dir, paste0("cox_reg_results_per_", ind_var, ".tsv")))
+  
+  print(table)
+  
+  # output the plot
+  pdf(file.path(plots_dir, paste0("coxph_survival_per_", ind_var, ".pdf")))
+  plot_coxph <- survminer::ggsurvplot(survfit(fit),
+                                      data=meta_indep,
+                                      xlim = c(0, 5000),
+                                      break.time.by = 500,
+                                      pval = TRUE, 
+                                      conf.int = TRUE,
+                                      risk.table = TRUE, # Add risk table
+                                      linetype = "strata", # Change line type by groups
+                                      surv.median.line = "hv", # Specify median survival
+                                      ggtheme = theme_bw())
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+
+  dev.off()
+
+  print(ggpar(plot_coxph[[1]], font.legend=6, legend="right"))
+  print(ggpar(plot_coxph[[2]], font.legend=6))
+}
+
+```
+
+## Log Rank analysis 
```

Comment:
This is great to keep around for the exploratory evidence stage. The interpretation of logrank can become challenging with more than two groups, like in `cancer_group_over10`.